components/test_coprocessor/dag.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+use protobuf::Message;
+use protobuf::RepeatedField;
+
+use kvproto::coprocessor::{KeyRange, Request};
+use kvproto::kvrpcpb::Context;
+use tipb::executor::{
+    Aggregation, ExecType, Executor, IndexScan, Limit, Selection, TableScan, TopN,
+};
+use tipb::expression::{ByItem, Expr, ExprType};
+use tipb::schema::ColumnInfo;
+use tipb::select::{Chunk, DAGRequest};
+
+use tikv::coprocessor::codec::{datum, table, Datum};
+use tikv::coprocessor::REQ_TYPE_DAG;
+use tikv::util::codec::number::NumberEncoder;
+
+pub struct DAGSelect {
+    pub execs: Vec<Executor>,
```

Comment:
Could you private these fileds?

Reply:
they are accessed in test_select directly